### PR TITLE
use const for  in relation to vault eventing metadata

### PR DESCRIPTION
**Pull request by hashiblaum**
### Description
What does this PR do?

Adds a const `path` for indicating the `path` metadata key used for in eventing

### CODE REVIEW

File: sdk/logical/events.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 
 // common event metadata keys
 const (
+	// EventMetadataPath is used in event metadata to show the API path the client must have the `subscribe` capability
+	// on in order consume the event. It is recommended that the event path metadata field uses that api path that was invoked
```
Comment (tvoran) [inline sdk/logical/events.go:16]:
```suggestion
	// on in order to consume the event. It is recommended that the event path metadata field uses the API path that was invoked
```

File: sdk/logical/events.go [hunk 2]
```diff
@@ -12,6 +12,10 @@ import (
...
+	// in order to generate the event
+	EventMetadataPath = "path"
```
Comment (tvoran) [inline sdk/logical/events.go:18]: Should we mention in the comment that the plugin mount will be prepended to this path?

Comment (hashiblaum) [reply sdk/logical/events.go]: yes! - updated the comments

File: sdk/logical/events.go [hunk 3]
```diff
@@ -12,10 +12,16 @@ import (
 
 // common event metadata keys
 const (
-	// EventMetadataDataPath is used in event metadata to show the API path that can be used to fetch any underlying
-	// data. For example, the KV plugin would set this to `data/mysecret`. The event system will automatically prepend
+	// EventMetadataPath is used in event metadata to show the API path the client must have the `subscribe` capability
+	// on in order consume the event. It is recommended that the event path metadata field is the API path that was invoked
```
Comment (tvoran) [inline sdk/logical/events.go:16]:
```suggestion
	// on in order to consume the event. It is recommended that the event path metadata field is the API path that was invoked
```

File: sdk/logical/events.go [hunk 4]
```diff
@@ -12,10 +12,16 @@ import (
...
+	// in order to generate the event.
+	//
+	// For example, the KV plugin would set this to `data/mysecret`. The event system will automatically prepend
 	// the plugin mount to this path, if present, so it would become `secret/data/mysecret`, for example.
 	// If this is an auth plugin event, this will additionally be prepended with `auth/`.
+	EventMetadataPath = "path"
+	// EventMetadataDataPath is used in event metadata to show the API path that can be used to fetch any underlying
+	// data. Similar to the `path` event metadata, the event system will automatically prepend the plugin mount to the `data_path`.
```
Comment (tvoran) [inline sdk/logical/events.go:24]: A couple of these lines are getting a little long; it looks like the doc strings were previously wrapped at column 120, so we should probably stick with that at least.

**Review by tvoran — APPROVED** [review]

**Review by tvoran — APPROVED** [review]

**Pull request merged**